solr/core/src/test/org/apache/solr/response/transform/TestChildDocTransformerHierarchy.java

```diff
@@ -136,12 +136,28 @@ public void testParentFilterLimitJSON() throws Exception {
   }
 
   @Test
-  public void testLiveDocs() throws Exception {
+  public void testWithDeletedChildren() throws Exception {
+    // add a doc to create another segment
+    final String addNonTestedDoc =
+        "{\n" +
+          "\"add\": {\n" +
+            "\"doc\": {\n" +
+                "\"id\": " + -1000 + ", \n" +
+                "\"type_s\": \"cake\", \n" +
+            "}\n" +
+          "}\n" +
+        "}";
+
     indexSampleData(numberOfDocsPerNestedTest);
     // delete toppings path
     assertU(delQ("_nest_path_:\\/toppings"));
     assertU(commit());
 
+    if (random().nextBoolean()) {
```

Comment:
Did you get this to trip the bug you had?  I'd be surprised/baffled if so.  The non-tested doc needs to be added _before_ the document you are testing so that you sometimes shift the document being tested into a segment that isn't the first one.  Otherwise `segBaseId` will always be 0 and won't expose the bug.  Make sense?